src/ui/statusline.c

```diff
@@ -373,6 +375,48 @@ parse_view_macros(view_t *view, const char **format, const char macros[],
 				LOG_INFO_MSG("Unmatched %%]");
 				ok = 0;
 				break;
+			case '{':
+				{
+					/* Try to find matching closing bracket
+					 * TODO: implement the way to escape it, so that the expr may contain
+					 * closing brackets */
+					const char *e = strchr(*format, '}');
+					char *expr = NULL, *resstr = NULL;
+					var_t res = var_false();
+					ParsingErrors parsing_error;
+
+					/* If there's no matching closing bracket, add the opening one
+					 * literally */
+					if (e == NULL) {
+						strappendch(&result, &len, '%');
+						strappendch(&result, &len, '{');
+						break;
+					}
+
+					/* Create a NULL-terminated copy of the given expr.
+					 * TODO: we could temporarily use buf for that, to avoid extra
+					 * allocation, but explicitly named variable reads better. */
+					expr = calloc(e - (*format) + 1 /* NUL-term */, 1);
+					memcpy(expr, *format, e - (*format));
+
+					/* Try to parse expr, and convert the res to string if succeed. */
+					parsing_error = parse(expr, &res);
+					if(parsing_error == PE_NO_ERROR) {
+						resstr = var_to_string(res);
+					}
+
+					if (resstr != NULL) {
+						copy_str(buf, sizeof(buf), resstr);
+					} else {
+						copy_str(buf, sizeof(buf), "<Invalid expr>");
+					}
```

Comment:
Style nitpicks to keep the style consistent (in tests too):
* should be no space after `if`
* curly braces need to be on separate lines

Reply:
Done.
However maintaining various style guidelines manually is annoying (especially when one got used to different conventions), I think it would be cool to employ clang-format, with the config which works for you, and pre-commit hook which would do all the magic. Combined with the CI job which would apply the clang-format and fail if there are differences, it would even eliminate the need to manually check style :)

Reply:
I'm not sure if clang-format is capable of handling the style, it's oriented towards big companies and their style guides. On the other hand this might be something to try to find out (maybe astyle will work), because I sometimes make style mistakes too :)